conjure-go-contract/errors/error_type_registry.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2020 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var registry = map[string]reflect.Type{}
+
+func RegisterErrorType(name string, typ reflect.Type) {
```

Comment:
Can we add documentation? In particular, should call out that registering duplicate value will panic, and would be helpful to have example to show how to get the `reflect.Type` required for this call.

Reply:
Yep the docs are coming here, just wanted to get this out so Adam could depend on it: https://github.com/palantir/conjure-go-runtime/pull/94/files#diff-e0d6df65a1193580fa3c60b479c01218R28